### PR TITLE
Adds a new INHALE reagent transfer method and applies it to smoking and smoke inhalation. Nitrous oxide (the reagent version) inhalation causes brain damage.

### PR DESCRIPTION
**Pull request by necromanceranne**
## About The Pull Request

Adds the ``INHALE`` transfer method. Smoking and smoke exposure now uses this method.

Makes a few sensible inhalation effects for current behaviours.

Adds a unit test amongst the rest of the exposure method tests.

Nitrous oxide does brain damage on inhalation based on volume inhaled.

## Why It's Good For The Game

Smoke and smoking being based on ingestion rather than inhaling was always weird to me. It'd be nice to have a specific method for smoking instead of a method shared by drinking. Many of our smokables make sense as indigestibles, obviously, but maybe we'd like a chem you can ONLY smoke.

~~Also, I walked out of a job interview and up the street to see a guy throwing his empty nitrous canisters around and yelling at passerbys, so shout out to that fucking guy with this PR.~~

## Changelog
:cl:
code: Introduces a INHALE method for reagent transfer. Cigarettes and smoke reactions use this new method.
balance: Nitrous oxide now brain damage if you inhale it. So don't do that.
/:cl:

### CODE REVIEW

**Comment by necromanceranne** [comment]
(Ill swap that to uuuh check hallucinations instead)

**Comment by SomeoneYouProbablyKnow** [comment]
Isn't that what we have in medbay anesthesia tanks? I'd be pretty bad if those damaged your brain. Plus N2O was the only (? i guess theres the healing one also) starting nonlethal gas.
All that for what's just a reference to haha funny galaxy gas

File: code/modules/mob/living/living_defense.dm [hunk 1]
```diff
@@ -622,15 +622,15 @@
...
+ * If the methods include INGEST or INHALE, the mob tastes the reagents.
  * If the methods include VAPOR it incorporates permiability protection.
  */
 /mob/living/expose_reagents(list/reagents, datum/reagents/source, methods=TOUCH, volume_modifier=1, show_message=TRUE)
 	. = ..()
 	if(. & COMPONENT_NO_EXPOSE_REAGENTS)
 		return
 
-	if(methods & INGEST)
+	if(methods & INGEST || methods & INHALE)
```
Comment (SyncIt21) [inline code/modules/mob/living/living_defense.dm:633]:
```suggestion
	if(methods & (INGEST | INHALE))
```

Comment (necromanceranne) [reply code/modules/mob/living/living_defense.dm:633]: oops, missed that

**Comment by necromanceranne** [comment]
> Isn't that what we have in medbay anesthesia tanks? I'd be pretty bad if those damaged your brain. Plus N2O was the only (? i guess theres the healing one also) starting nonlethal gas. All that for what's just a reference to haha funny galaxy gas

Breathing N2O gas does not inject the reagent, which does completely separate effects to the gas.

**Comment by LSommers** [comment]
> > Isn't that what we have in medbay anesthesia tanks? I'd be pretty bad if those damaged your brain. Plus N2O was the only (? i guess theres the healing one also) starting nonlethal gas. All that for what's just a reference to haha funny galaxy gas
> 
> Breathing N2O gas does not inject the reagent, which does completely separate effects to the gas. They are not related, and probably shouldn't be.

It's kinda bizarre that N2O made in chem can become a smoke (gas) that's completely different from atmospheric N2O, come to think of it. Especially when spilling that same chem N2O... turns it into atmos N2O and not the N2O it would become if smoke reactioned???

**Comment by SomeoneYouProbablyKnow** [comment]
> Breathing N2O gas does not inject the reagent, which does completely separate effects to the gas. They are not related, and probably shouldn't be.

Good. Completely bizarre, i don't think i even knew we had a chem for it and that it was its own thing. But good.

**Comment by Loafin34** [comment]
can you add the nitrous reagent to people that get sprayed in the face with spraycans?

**Comment by Blacklist897** [comment]
I think the blast off ampoule should be made to use inhale, because the way its described and presented it looks like something you inhale

**Comment by BlueMemesauce** [comment]
Maybe something other than inhale like "Smoked" would make it more clear?

**Comment by OverwatchVoice** [comment]
> > Isn't that what we have in medbay anesthesia tanks? I'd be pretty bad if those damaged your brain. Plus N2O was the only (? i guess theres the healing one also) starting nonlethal gas. All that for what's just a reference to haha funny galaxy gas
> 
> Breathing N2O gas does not inject the reagent, which does completely separate effects to the gas. They are not related, and probably shouldn't be.

I feel like you should make that difference very clear in the changelog / PR title. Otherwise it sounds like a malicious misnomer.

**Comment by nikothedude** [comment]
oh hey this is the thing i was trying to do a while ago with asthma

**Comment by necromanceranne** [comment]
> oh hey this is the thing i was trying to do a while ago with asthma

Lol someone sent me that PR and I hadn't realized you had already given this a go already.

**Comment by necromanceranne** [comment]
> Reference to galaxy gas isn't worth screwing over my beloved N20

Okay so, how?

**Comment by necromanceranne** [comment]
> > > Isn't that what we have in medbay anesthesia tanks? I'd be pretty bad if those damaged your brain. Plus N2O was the only (? i guess theres the healing one also) starting nonlethal gas. All that for what's just a reference to haha funny galaxy gas
> > 
> > 
> > Breathing N2O gas does not inject the reagent, which does completely separate effects to the gas. They are not related, and probably shouldn't be.
> 
> I feel like you should make that difference very clear in the changelog / PR title. Otherwise it sounds like a malicious misnomer.



> I think the blast off ampoule should be made to use inhale, because the way its described and presented it looks like something you inhale

That's a good point.

**Pull request merged**